File: Upcoder/Other/Kiểm tra Số nguyên tố/LTCB-93.cs
using System;

namespace CheckSNT
{
    public class Program
    {
        public static void Main(string[] args)
        {
            long n;
            string input;
            while (!string.IsNullOrEmpty(input = Console.ReadLine()))
            {
                if (long.TryParse(input, out n))
                {
                    if (CheckSNT(n))
                        Console.WriteLine("Yes");
                    else
                        Console.WriteLine("No");
                }
                else
                {
                    Console.WriteLine();
                }
            }
        }

        public static bool CheckSNT(long n)
        {
            if (n < 2)
                return false;
            if (n == 2 || n == 3)
                return true;
            if (n % 2 == 0 || n % 3 == 0)
                return false;

            for (long i = 5; i * i <= n; i += 6)
            {
                if (n % i == 0 || n % (i + 2) == 0)
                    return false;
            }
            return true;
        }
    }
}

File: Upcoder/Other/Kiểm tra Số nguyên tố/LTCB-93.py
def check_snt(n):
    if n < 2:
        return False
    if n == 2 or n == 3:
        return True
    if n % 2 == 0 or n % 3 == 0:
        return False
    
    i = 5
    while i * i <= n:
        if n % i == 0 or n % (i + 2) == 0:
            return False
        i += 6
    return True

def main():
    while True:
        try:
            n = int(input())
            if check_snt(n):
                print("Yes")
            else:
                print("No")
        except EOFError:
            break  # Dừng lại khi không còn dữ liệu đầu vào

if __name__ == "__main__":
    main()
